### PR TITLE
GH-119: Improve readme

### PR DESCRIPTION
**Pull request by CMDJojo**
This is a PR aimed at general improvements to the README, which aims to give a more welcoming and helpful presentation of the project. It includes an overview of the project, a breakdown of its components, links to those, and some improvements of the documentation of each component. It also includes our license and the licenses of the third-party packages we are using.

It is in no way perfect but is a good idea to merge before the "halvtidspresentation", where people may want to check out our project. Since it includes no code changes, I think it is okay to merge even though we have a code freeze, and it shouldn't break anything since nothing is depending on the readme files.

Resolves GH-119

### CODE REVIEW

File: MODMARK.md [hunk 1]
```diff
@@ -86,4 +79,49 @@ Example 5: module with arguments
...
+
+| Source                | Output                      |
+|-----------------------|-----------------------------|
+| ```"double quotes"``` | &ldquo;double quotes&ldquo; |
+| ```'single quotes'``` | &lsquo;single quotes&rsquo; |
+| ```a -- b```          | a &ndash; b                 |
+| ```a --- b```         | a &mdash; b                 |
+| ```ellipsis...```     | ellipsis&hellip;            |
+
+*Note: character sequences that could technically be broken up into smaller sequences for smart punctuation will instead be used as is. This means that `----` is not parsed as two endashes, but rather one emdash followed by a normal dash.*
```
Comment (Forsinge) [inline MODMARK.md:127]: I'm not sure this is how we actually implemented it. I think the result here is simply four dashes.

File: MODMARK.md [hunk 2]
```diff
@@ -11,36 +10,28 @@ Tags and smart punctuation are also part of the language. Information about thes
...
+Modules are programs that are run from within your document, transforming the text you give them at their discretion. To use, or *invoke*, a module, encase its name within square brackets, and place the text you want to pass to it afterwards: `[link] https://example.com`. The module, in this case `link`, is invoked, consuming the text you give it, `https://example.com`.
 
 There are two ways to invoke a module, **inline** or **multiline**. When invoking a module in the middle of a paragraph, you do it **inline**. The module will consume the text following it, until the next space or until the end of the line, so you can write `Then, you take [math] x^2 and find out ...`. If you invoke the module **multiline**, you write it as if it was its own paragraph, with one empty row before and after it. It may then consume multiple rows. Here is one example of that:
 
+**Example: Mutliline Module**
```
Comment (Forsinge) [inline MODMARK.md:34]: Spelling

File: MODMARK.md [hunk 3]
```diff
@@ -11,36 +10,28 @@ Tags and smart punctuation are also part of the language. Information about thes
 
 ## Paragraphs
 
-Paragraphs are sequences of text separated by one or more empty rows. A single backslash is used at the end of a row to join the following row in the compiled text. Double backslash is used to include a break in the compiled text without creating a new paragraph.
-
-Example 1: simple paragraphs
+Each paragraph consist of one or more lines of text, and ends by one or many blank rows, similar to markdown. Line breaks will be kept and may or may not be rendered by the output format. Many output formats such as HTML, Markdown and LaTeX doesn't render line breaks. If you want to escape the line break to ensure that it is outputted on the same line, you can put a backslash at the end of the line. Here is an example of some paragraphs:
 
     This is the first sentence of the first paragraph.
```
Comment (Forsinge) [inline MODMARK.md:15]: Maybe we can explicitly label this example as well to be consistent with the others, or maybe even remove all labels.

**Review by Forsinge — CHANGES_REQUESTED** [review]
Very good! It's a big improvement from the previous documents. I commented on a few very minor things I picked up while reading.

File: MODMARK.md (hunk 3 above)
Comment (CMDJojo) [reply MODMARK.md:15]: Fixed!

File: MODMARK.md (hunk 2 above)
Comment (CMDJojo) [reply MODMARK.md:34]: Fixed!

File: MODMARK.md (hunk 1 above)
Comment (CMDJojo) [reply MODMARK.md:127]: Fixed!

File: MODMARK.md [hunk 4]
```diff
@@ -86,4 +79,49 @@ Example 5: module with arguments
...
+    The equation [math style=italic] x^2 is simple enough, but for [math style=bold]!x^2 + y^3!, you'll need to study multivariable calculus
+
+## Escaping characters
+
+You may escape the upcoming character by using a backslash. The escaped character will be rendered as-is, and may not be treated as a part of a module, tag, escape character or smart punctuation. The only exception to this is a backslash just before a line break. In that case, the line break gets removed.
+
+## Tags
+
+Similarly to Markdown, ModMark supports tags. Tags are character sequences that may encase text and/or inline modules, and is used to apply additional formatting. An example of one such tag is `**`, which makes the encased text bold: `**abc**` will be rendered as **abc**. All opening and closing tags in the language by default consists of one symbol repeated two times. Note that more tags may be added in upcoming releases, and upcoming releases may also support customized tags. Some tags allow nesting: `**bold //italic and bold//**` will be rendered as **bold *italic and bold***, while some tags, like <code>\``verbatim``</code> doesn't allow for nested tags. Here are the tags supported by default:
```
[the remark this answers is not in the document]
Comment (CMDJojo) [reply MODMARK.md:90]: I think "doesn't" is better since it refers to "some tags" in plural.

[the remark this answers is not in the document]
Comment (CMDJojo) [reply MODMARK.md:90]: Added an additional comma to clarify that it refers to "some tags" rather to "verbatim" directly

File: MODMARK.md (hunk 1 above)
Comment (CMDJojo) [reply MODMARK.md:127]: I changed this to a text explaining what is actually happening. You are correct in that four dashes becomes four dashes

**Comment by CMDJojo** [comment]
Thank you for your review @Forsinge , thanks for pointing out those mistakes! I have now responded to your comments and force-pushed my changes, and I would be happy if you would look at it again and see if you find something else that is off.

File: MODMARK.md [hunk 5]
```diff
@@ -53,9 +44,9 @@ This code defines ...
...
+This works fine most of the time, but sometimes you may want to include multiple words to a module you invoke. This is the place where *delimiters* come in. Immediately following the square brackets, you may define your own *opening delimiter*. If you do, all text following up until the matching *closing delimiter* is taken, like this: `... a more complex equation like [math](x^2 + y^2 + z^2) may be ...`. In this case, the module gets passed all the text `x^2 + y^2 + z^2`, including the spaces but not including the brackets. You may use any non-alphanumeric character as the opening delimiter, and if you use brackets (like in the example above), the matching closing delimiter has the opposing bracket used. For **inline** modules, one character is allowed as the delimiter, and for **multiline ** modules, any amount of characters is allowed. For a multiline module, the opening delimiter `{{(` is matched by the closing delimiter `)}}`. Here is some more examples of delimiter usage:
```
[the remark this answers is not in the document]
Comment (CMDJojo) [reply MODMARK.md]: Fixed!

**Pull request merged**